### ISSUE TITLE
How to enable syntax highlight/decoration etc?

### ISSUE THREAD

**Issue by rougsig** [BurningTreeC/tiddlywiki-codemirror-6#2]
Hi!

Looks like it exists in the repo, but does not work?

https://github.com/BurningTreeC/tiddlywiki-codemirror-6/blob/main/tiddlywiki-codemirror-6/files/modules/parser/parser.js

![image](https://github.com/user-attachments/assets/aa69a8a0-59eb-458e-840c-922f4be5fca5)

TY

**Comment by BurningTreeC** [comment]
Hi! Oh you catched me during a biiiiig WIP!
Just wait until it's done :)

**Comment by rougsig** [comment]
Oh got it. Im working on it too

Here https://github.com/rougsig/tiddlywikix

**Comment by BurningTreeC** [comment]
Yes, I've found out that integrating it directly into lezer might not be a good approach since lezer is an incremental parser and I want to parse the full text using $tw.wiki.parseText("text/vnd.tiddlywiki", text, ... ).tree
And tokenize the tree

**Comment by rougsig** [comment]
But as i understand we need to use lezer common package for api integration with codemirror. How are you plan to implement it without that api?

**Comment by BurningTreeC** [comment]
I use a custom ViewPlugin that has an updateListener and on each update, when the doc has changed, it parses the full text.
Then it tokenizes the text and gives back decorations, there are also autocompletions.
But at the moment only the API is working, the tokenization is a big work

**Comment by rougsig** [comment]
Interesting approach. Is it laggy?

**Comment by BurningTreeC** [comment]
I still have to find that out

**Comment by BurningTreeC** [comment]
I think using lezer is not the right approach. Because it brings in chunks of text ... we can work around that and always create the full text and parse that, I have another approach doing so, but that seemed way too complicated to me

**Comment by rougsig** [comment]
> I think using lezer is not the right approach. Because it brings in chunks of text ... we can work around that and always create the full text and parse that, I have another approach doing so, but that seemed way too complicated to me

As i understand input is always the full text in the editor. From lezer markdown they are parse full text, then do updates on thee based on regions

**Comment by BurningTreeC** [comment]
> > I think using lezer is not the right approach. Because it brings in chunks of text ... we can work around that and always create the full text and parse that, I have another approach doing so, but that seemed way too complicated to me
> 
> As i understand input is always the full text in the editor. From lezer markdown they are parse full text, then do updates on thee based on regions

Yes, you can get the full text from the input
